### PR TITLE
Strongly typed coordinates

### PR DESCRIPTION
**Pull request by evil159**
### What does this pull request do?

* Bumps Pigeon to 17.1.2
* Replaces `Map<String?, Object?>?` with `Point` from [turf](https://pub.dev/packages/turf) package.

### What is the motivation and context behind this change?

Improve ergonomics of API usage.

### Pull request checklist:
 - [x] Add a changelog entry.
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Add documentation comments for any added or updated public APIs.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
+### main
+
+#### ⚠️ Breaking change
+
+`Map<String?, Object?>?` is replaced wiht `Point` from [turf](https://pub.dev/packages/turf) package.
```
Comment (maios) [inline CHANGELOG.md:5]:
```suggestion
`Map<String?, Object?>?` is replaced with `Point` from [turf](https://pub.dev/packages/turf) package.
```

Comment (maios) [reply CHANGELOG.md:5]: Can we specify link to the `Point` class https://pub.dev/documentation/turf/latest/turf/Point-class.html

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+CameraOptions(
+    center: Point(
+        coordinates: Position(
+        -0.11968,
+        51.50325,
+    )))
+```
+
+* Bump Pigeon to 17.1.2
+* Replace `Map<String?, Object?>?` with `Point` from [turf](https://pub.dev/packages/turf) package.
```
Comment (maios) [inline CHANGELOG.md:27]:
```suggestion
```
this is duplicated line

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/mapping/turf/TurfAdapters.kt [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+fun Point.toList(): List<Any?> {
+  return listOf(mapOf("coordinates" to coordinates()))
+}
+
+object PointDecoder {
+  fun fromList(list: List<Any?>): Point {
+    return Point.fromJson(Gson().toJson(list.first()))
+  }
+}
```
Comment (maios) [inline android/src/main/kotlin/com/mapbox/maps/mapbox_maps/mapping/turf/TurfAdapters.kt:14]: let's add newline at EOF?

File: ios/Classes/Generated/CircleAnnotationMessager.swift [hunk 4]
```diff
@@ -72,23 +72,23 @@ struct CircleAnnotation {
...
+  var circleRadius: Double?
   /// The stroke color of the circle.
-  var circleStrokeColor: Int64? = nil
+  var circleStrokeColor: Int64?
   /// The opacity of the circle's stroke.
-  var circleStrokeOpacity: Double? = nil
+  var circleStrokeOpacity: Double?
   /// The width of the circle's stroke. Strokes are placed outside of the `circle-radius`.
-  var circleStrokeWidth: Double? = nil
+  var circleStrokeWidth: Double?
```
Comment (maios) [inline ios/Classes/Generated/CircleAnnotationMessager.swift:91]: are these `= nil` needed? or is this just how Pigeon 17.1.2 is?

**Review by maios — APPROVED** [review]

File: android/src/main/kotlin/com/mapbox/maps/mapbox_maps/mapping/turf/TurfAdapters.kt (hunk 3 above)
Comment (evil159) [reply android/src/main/kotlin/com/mapbox/maps/mapbox_maps/mapping/turf/TurfAdapters.kt:14]: Ah, no can do, ktlint doesn't like trailing newlines 🤷

**Pull request merged**